src/Microsoft.TemplateEngine.Abstractions/ITemplateMetadata.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions.Constraints;
+using Microsoft.TemplateEngine.Abstractions.Parameters;
+
+namespace Microsoft.TemplateEngine.Abstractions
+{
+    public interface ITemplateMetadata
```

Comment:
hopefully we would base `ITemplateInfo` and `ITemplate` on this interface and `ITemplateLocator` interfaces at some point to make them more granular.